lib/grandpa/message_handler.go

```diff
@@ -299,20 +299,43 @@ func getEquivocatoryVoters(votes []AuthData) map[ed25519.PublicKeyBytes]struct{}
 	return eqvVoters
 }
 
+func verifyIfDescendantOfHighestFinalisedBlock(blockState BlockState, hash common.Hash) error {
```

Comment:
Why not use this as a method of `*MessageHandler` and use `h.blockState` instead of passing the reference of the `BlockState`?
```suggestion
func (h *MessageHandler) verifyIfDescendantOfHighestFinalisedBlock(hash common.Hash) error {
```

Reply:
It is used in service.VerifyBlockJustification as well! If I create a method I can't use `verifyIfDescendantOfHighestFinalisedBlock` there